adapters/mqtt-vertx-base/src/test/java/org/eclipse/hono/adapter/mqtt/AbstractVertxBasedMqttProtocolAdapterTest.java

```diff
@@ -1363,6 +1425,8 @@ public void verifyEventMessageUsesTtlValueGivenInPropertyBag(final VertxTestCont
         forceClientMocksToConnected();
 
         final DownstreamSender sender = mock(DownstreamSender.class);
+        when(sender.sendAndWaitForOutcome(any(Message.class), (SpanContext) any()))
+                .thenReturn(Future.failedFuture("some error"));
```

Comment:
Previously, without this, there was an NPE thrown and caught via a Future.compose block here. Now, because of changed Future composition, this wouldn't get caught anymore. Therefore making it explicit that the test expects `sendAndWaitForOutcome` to fail here.